hid-uclogic-rdescs.c

```diff
@@ -0,0 +1,665 @@
+#include <linux/device.h>
+#include "hid-uclogic-rdescs.h"
+
+/* Fixed WP4030U report descriptor */
+const __u8 uclogic_wp4030u_rdesc_fixed[] = {
+        0x05, 0x0D,         /*  Usage Page (Digitizer),             */
+        0x09, 0x02,         /*  Usage (Pen),                        */
+        0xA1, 0x01,         /*  Collection (Application),           */
+        0x85, 0x09,         /*      Report ID (9),                  */
+        0x09, 0x20,         /*      Usage (Stylus),                 */
+        0xA0,               /*      Collection (Physical),          */
+        0x75, 0x01,         /*          Report Size (1),            */
+        0x09, 0x42,         /*          Usage (Tip Switch),         */
+        0x09, 0x44,         /*          Usage (Barrel Switch),      */
+        0x09, 0x46,         /*          Usage (Tablet Pick),        */
+        0x14,               /*          Logical Minimum (0),        */
+        0x25, 0x01,         /*          Logical Maximum (1),        */
+        0x95, 0x03,         /*          Report Count (3),           */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0x95, 0x05,         /*          Report Count (5),           */
+        0x81, 0x01,         /*          Input (Constant),           */
+        0x75, 0x10,         /*          Report Size (16),           */
+        0x95, 0x01,         /*          Report Count (1),           */
+        0x14,               /*          Logical Minimum (0),        */
+        0xA4,               /*          Push,                       */
+        0x05, 0x01,         /*          Usage Page (Desktop),       */
+        0x55, 0xFD,         /*          Unit Exponent (-3),         */
+        0x65, 0x13,         /*          Unit (Inch),                */
+        0x34,               /*          Physical Minimum (0),       */
+        0x09, 0x30,         /*          Usage (X),                  */
+        0x46, 0xA0, 0x0F,   /*          Physical Maximum (4000),    */
+        0x26, 0xFF, 0x7F,   /*          Logical Maximum (32767),    */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0x09, 0x31,         /*          Usage (Y),                  */
+        0x46, 0xB8, 0x0B,   /*          Physical Maximum (3000),    */
+        0x26, 0xFF, 0x7F,   /*          Logical Maximum (32767),    */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0xB4,               /*          Pop,                        */
+        0x09, 0x30,         /*          Usage (Tip Pressure),       */
+        0x26, 0xFF, 0x03,   /*          Logical Maximum (1023),     */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0xC0,               /*      End Collection,                 */
+        0xC0                /*  End Collection                      */
```

Comment:
The Linux coding guidelines call for using TABs for indentation. Please replace the spaces with TABs in the initial indentation of this and following report descriptors, so that they match the original code in hid-uclogic.c. You can checkout the exact changes between the files with e.g. `vimdiff <(git show origin/master:hid-uclogic.c) hid-uclogic-rdescs.c)` and edit the file to match at the same time.